lib/vtls/mbedtls.c

```diff
@@ -343,7 +343,8 @@ mbed_connect_step1(struct connectdata *conn,
   if(SSL_SET_OPTION(key)) {
     ret = mbedtls_pk_parse_keyfile(&BACKEND->pk, SSL_SET_OPTION(key),
                                    SSL_SET_OPTION(key_passwd));
-    if(ret == 0 && !mbedtls_pk_can_do(&BACKEND->pk, MBEDTLS_PK_RSA))
+    if(ret == 0 && !(  mbedtls_pk_can_do(&BACKEND->pk, MBEDTLS_PK_RSA)
+                    ||  mbedtls_pk_can_do(&BACKEND->pk, MBEDTLS_PK_ECKEY)))
```

Comment:
Is this `MBEDTLS_PK_ECKEY` flag supported on all mbedtls versions or is there a risk that this will break the build for some?

Reply:
A quick skim of the code seems to imply that usage of `MBEDTLS_PK_ECKEY` depends on `MBEDTLS_ECP_C`.

Reply:
> Is this `MBEDTLS_PK_ECKEY` flag supported on all mbedtls versions or is there a risk that this will break the build for some?

If you mean historically, I think it dates back from the polarssl days : https://github.com/ARMmbed/mbedtls/commit/211a64c79f496de1a82083f517230e7fd1b009fd

So it is probably have been available in all mbedtls versions.

Reply:
Yes confirmed, It then have been renamed to the actual name - without polarssl reference - in mbedtls release 2.0  : https://github.com/ARMmbed/mbedtls/blob/4cb87f409df0ddd878ea50cfca7dc8735ee574f2/include/mbedtls/pk.h#L74